sdc/datatypes/hpat_pandas_dataframe_functions.py

```diff
@@ -1867,13 +1867,63 @@ def _df_getitem_unicode_idx_impl(self, idx):
     ty_checker.raise_exc(idx, expected_types, 'idx')
 
 
+def df_getitem_single_label_loc_codegen(self, idx):
+    """
+    Example of generated implementation:
+        def _df_getitem_single_label_loc_impl(self, idx):
+            for i in numba.prange(len(self._dataframe.index)):
+                if self._dataframe._index[i] == idx:
+                data_0 = pandas.Series(self._dataframe._data[0], index=self._dataframe.index)
+                result_0 = data_0.at[idx]
+                data_1 = pandas.Series(self._dataframe._data[1], index=self._dataframe.index)
+                result_1 = data_1.at[idx]
+                return pandas.Series(data=[result_0[0], result_1[0]], index=['A', 'B'], name=str(idx))
+            raise IndexingError('Index is out of bounds for axis')
+    """
+    func_lines = ['def _df_getitem_single_label_loc_impl(self, idx):',
+                  '  for i in numba.prange(len(self._dataframe.index)):',
+                  '    if self._dataframe._index[i] == idx:']
+    if isinstance(self.index, types.NoneType):
+        func_lines = ['def _df_getitem_single_label_loc_impl(self, idx):',
+                      '  if -1 < idx < len(self._dataframe._data):']
+    results = []
+    result_index = []
+    for i, c in enumerate(self.columns):
+        result_c = f"result_{i}"
+        func_lines += [f"      data_{i} = pandas.Series(self._dataframe._data[{i}], index=self._dataframe.index)",
+                       f"      {result_c} = data_{i}.at[idx]"]
+        results.append(result_c)
+        result_index.append(c)
+    data = '[0], '.join(col for col in results) + '[0]'
+    func_lines += [f"      return pandas.Series(data=[{data}], index={result_index}, name=str(idx))",
+                   f"  raise IndexingError('Index is out of bounds for axis')"]
+
+    func_text = '\n'.join(func_lines)
+    global_vars = {'pandas': pandas, 'numpy': numpy,
+                   'numba': numba,
+                   'IndexingError': IndexingError}
+
+    return func_text, global_vars
+
+
+gen_df_getitem_loc_single_label_impl = gen_impl_generator(
+    df_getitem_single_label_loc_codegen, '_df_getitem_single_label_loc_impl')
+
+
 @sdc_overload(operator.getitem)
 def sdc_pandas_dataframe_accessor_getitem(self, idx):
     if not isinstance(self, DataFrameGetitemAccessorType):
         return None
 
     accessor = self.accessor.literal_value
 
+    if accessor == 'loc':
+        if isinstance(idx, types.Integer):
+            return gen_df_getitem_loc_single_label_impl(self.dataframe, idx)
+
+        raise TypingError('Operator getitem(). The index must be a single label, a  list or array of labels,\
+                          a slice object with labels, a boolean array or a callable. Given: {}'.format(idx))
```

Comment:
Is `getitem()` correct? Shouldn't we use this message info in `limitations` block in docstring?

Reply:
What's about this recommendation?

Reply:
This must be so that the user understands that he entered something incorrect
The same is done in the series.

Reply:
I meant:
1. Actually the operator is not `getitem`, it's `loc`.
2. `The index must be a single label, a  list or array of labels, a slice object with labels, a boolean array or a callable` looks like a limitation, doesn't?

Reply:
How to fix it then?

Reply:
```suggestion
        ty_checker = TypeChecker('Operator loc().')
        ty_checker.raise_exc(idx, 'int', 'idx')
```

I meant to insert limitations block to docstring as it was done in https://github.com/IntelPython/sdc/commit/0e1ce3a124d428737f7e2e0e98cd739d574dcf0e#diff-37d3d013a811f054d85ea0713b88b1eeR1723-R1731. However `idx` can be only of integer according to the code.

Reply:
I already do that
    - Loc works with basic case only: single label
in limitations

Reply:
You still raise exception with incorrect message.

Reply:
Fixed